<a href="https://colab.research.google.com/github/Stephanie-Jarrell/WISC-DS-02_PYTHON/blob/main/5_4_Async_Stephanie_Jarrell.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [44]:
from google.colab import files
uploaded = files.upload()

Saving music_genre_classification.csv to music_genre_classification.csv


In [8]:
import pandas as pd

In [9]:
import plotly.express as px

In [45]:
'''
LOUNGER / QUIET TYPES AUDIENCE
'''
# Read CSV for the column you need for lounger audience
df = pd.read_csv("music_genre_classification.csv", usecols = ['Artist Name', 'energy', 'loudness', 'acousticness'])

# Get the smallest 10 energy values
low_energy = df.nsmallest(10, 'energy')
print(low_energy[['Artist Name', 'energy']])


# Plot a bar chart, using plotly
px.bar(low_energy, 'Artist Name', 'energy', title='10 Lowest Energy Songs by Artist', color='energy', color_continuous_scale='Greens')

                                            Artist Name   energy
6769                               Todd Anderson-Kunert  0.00002
3943         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma  0.00433
2386  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabro...  0.00526
2758                                              HILMA  0.00700
3944  Luigi Boccherini, Tim Hugh, Scottish Chamber O...  0.00714
2469                              Gerry Mulligan Sextet  0.00894
1028                                 Broccoli, you too?  0.00976
2326             Francisco Villegas, Julio Mariñelarena  0.01040
3770                                      Library Tapes  0.01080
1169                                        Carlos Cipa  0.01230


In [11]:
# Get the lowest 10 loudness values in decibels
quietest = df.nsmallest(10, 'loudness')
print(quietest[['Artist Name', 'loudness']])


# Plot to a bar chart in plotly
px.bar(quietest, 'Artist Name', 'loudness', title='10 Quietest Songs by Artist', color='loudness', color_continuous_scale='blues')

                                            Artist Name  loudness
3561  Luigi Boccherini, Tim Hugh, Scottish Chamber O...   -36.544
2801         Ludwig van Beethoven, Emanuel Ax, Yo-Yo Ma   -32.639
1822  Alexander Glazunov, Alexander Rudin, Moscow Ph...   -32.520
3150                      Victor Kozlov, Irina Kulikova   -31.846
1268  Johann Sebastian Bach, Freiburger Barockorchester   -30.625
7432                                              HILMA   -29.955
1521  G. I. Gurdjieff, Anja Lechner, Vassilis Tsabro...   -29.845
6651              Ottorino Respighi, Magdalena Hoffmann   -29.234
3997                                        Carlos Cipa   -29.161
4895                                      Tony Anderson   -28.958


In [18]:
# Get the highest 10 acousticness values
best_acoustics = df.nlargest(10, 'acousticness')
print(best_acoustics[['Artist Name', 'acousticness']])

# Plot to a bar chart in plotly
px.bar(best_acoustics, 'Artist Name', 'acousticness', title = '10 Best Acoustic Songs by Artist', color = 'acousticness', color_continuous_scale='sunsetdark')


                           Artist Name  acousticness
757        Memphis Minnie & Kansas Joe         0.995
1264                  Blind Boy Fuller         0.995
3206                   Baby Boy Warren         0.995
5244  Anja Lechner, François Couturier         0.995
6349                     Raquel Meller         0.995
1404                  Daniel Raticelli         0.994
2124                     Juanita Reina         0.994
3649                   Thelonious Monk         0.994
4237              Antonio Garcia Isaac         0.994
6154                        R.J. Smith         0.994


12. Run that cell and see your small table print and bar chart plot. What do you notice about this bar chart that’s different from the others? Why do you think it’s different?

A: The color scale is just two colors because the acusticness values are all 0.995 or 0.994.  The color scale and the Y axes scale do not showcase this data the best.  


In [19]:
# Get the highest 30 acousticness values
best_acoustics = df.nlargest(30, 'acousticness')
print(best_acoustics[['Artist Name', 'acousticness']])

# Plot to a bar chart in plotly
px.bar(best_acoustics, 'Artist Name', 'acousticness', title = '10 Best Acoustic Songs by Artist', color = 'acousticness', color_continuous_scale='sunsetdark')

                                      Artist Name  acousticness
757                   Memphis Minnie & Kansas Joe         0.995
1264                             Blind Boy Fuller         0.995
3206                              Baby Boy Warren         0.995
5244             Anja Lechner, François Couturier         0.995
6349                                Raquel Meller         0.995
1404                             Daniel Raticelli         0.994
2124                                Juanita Reina         0.994
3649                              Thelonious Monk         0.994
4237                         Antonio Garcia Isaac         0.994
6154                                   R.J. Smith         0.994
6426  Franz Schubert, Mischa Maisky, Daria Hovora         0.994
6700                                 Diane Edison         0.994
630                            Juanito Valderrama         0.993
688                               Lata Mangeshkar         0.993
893             Rekha, Vinod Khanna, Ram


13. Now try changing the 10 value in the nlargest() function to 30, then run the cell again. Is it more clear why the bar chart was different? Explain

A: While the bar are still extremley similar hights due to the Y axis scale, the colors vary more than the 10 value chart.  This shows a wider range of acouticness. 

#Track Team / Partiers

* high energy - nlargest 'energy'
* high tempo - nlargest 'tempo'
* short duration - nsmallest 'duration_in min/ms'


In [61]:
'''
Track Team / Partiers
'''
# Read CSV for the column you need for lounger audience
df2 = pd.read_csv("music_genre_classification.csv", usecols = ['Artist Name', 'Track Name', 'energy', 'tempo', 'duration_in min/ms'])

# Get the largest 10 energy values
high_energy = df2.nlargest(10, 'energy')
print(high_energy[['Artist Name', 'energy']])


# Plot a bar chart, using plotly
px.bar(high_energy, 'Artist Name', 'energy', title='10 Highest Energy Songs by Artist', color='energy', color_continuous_scale='bluered')

# The visuals showed I had duplicates, so I went back to cleaned up the CSV and re-uploaded

                  Artist Name  energy
1531               Darkthrone   1.000
2217                FEVER 333   0.998
3646                     Lard   0.998
2736               Hellhammer   0.997
2739               Hellripper   0.997
5025           Pretty Vicious   0.997
6184        The Acacia Strain   0.997
38             A Lethal Smile   0.996
1659              Destruction   0.996
2241  Five Finger Death Punch   0.996


In [57]:
# Get the highest 10 highest tempo values
tempo = df2.nlargest(10, 'tempo')
print(tempo[['Artist Name', 'tempo']])


# Plot to a bar chart in plotly
px.bar(tempo, 'Artist Name', 'tempo', title='10 Highest Tempo by Artist', color='tempo', color_continuous_scale='tempo')


#I had to use the color scale called tempo, although I don't think it really matches the vibe of high tempo

            Artist Name    tempo
2852                IDK  249.438
4521         New Candys  216.194
2362         frown line  216.091
276   Angels & Airwaves  216.020
49      Ólafur Arnalds  212.849
118            AFN Peso  212.166
2134               Evil  209.657
133         Aisha Badru  208.648
3254      Johnny Winter  208.576
5842  Souls Of Mischief  206.247


In [80]:
# Get the shortest duration values in min/ms
duration = df2.nsmallest(10, 'duration_in min/ms')
print(duration[['Artist Name', 'Track Name', 'duration_in min/ms']])


# Plot to a bar chart in plotly
px.bar(duration, 'Track Name', 'duration_in min/ms', title='10 Highest Tempo by Track', color='duration_in min/ms', color_continuous_scale='curl')

# I changed the graph to Track instead of Artist b/c one artist had two tracks there were being stacked in the graph.  It wasn't a data error, but I didn't know how to fix it.  I wanted the graph to show Track & Artist, but I couldn't get that to work.  


                                Artist Name  \
4023                        Mahendra Kapoor   
5106                           R. D. Burman   
2865             Ilaiyaraaja, K. J. Yesudas   
1960   Eleni Karaindrou, Alexandros Botinis   
3411                   Kavita Krishnamurthy   
5247                          Reyanna Maria   
5105                           R. D. Burman   
593                            Bappi Lahiri   
4054  Mani Sharma, Amit Heri, Rajesh Vaidya   
3613                               Labrinth   

                                       Track Name  duration_in min/ms  
4023                           Jahan Daal Daal Pe            0.687450  
5106                   The Bullet Train - Reprise            1.209367  
2865         Surmayee Ankhiyon Mein - Sad Version            1.291667  
1960                   Je ne me consolerai jamais            1.353333  
3411                Zindagi Ki Yahi Reet Hai(Sad)            1.426883  
5247                                    So Pretty 

# NYT bestseller book club / literary fans

* high popluarity - nlargest 'Popularity'
* high danceability - nlargest 'danceability'
* high speechiness - nlargest 'speechiness'

In [83]:
'''
NYT bestseller book club / literary fans
'''
# Read CSV for the column you need for lounger audiencein
df3 = pd.read_csv("music_genre_classification.csv", usecols = ['Artist Name', 'Track Name', 'Popularity', 'danceability', 'speechiness'])

# Get the largest 10 popular values
popular = df3.nlargest(10, 'Popularity')
print(popular[['Artist Name', 'Popularity']])


# Plot a bar chart, using plotly
px.bar(popular, 'Artist Name', 'Popularity', title='10 Most Popular Songs by Artist', color='Popularity', color_continuous_scale='purd')


                  Artist Name  Popularity
3799                Lil Nas X       100.0
3340            Justin Bieber        99.0
1740                 Doja Cat        98.0
1874               Ed Sheeran        97.0
4140              Masked Wolf        97.0
552                 Bad Bunny        96.0
1048               Bruno Mars        96.0
2973  J Balvin, Maria Becerra        95.0
6660               The Weeknd        94.0
757             Billie Eilish        93.0


In [86]:
# Get the largest 10 dancable values
dancable = df3.nlargest(10, 'danceability')
print(dancable[['Artist Name', 'danceability']])


# Plot a bar chart, using plotly
px.bar(dancable, 'Artist Name', 'danceability', title='10 Most Dancable Songs by Artist', color='danceability', color_continuous_scale='rainbow')

# way too much color, but I'm having fun going through all the built in color scales

                   Artist Name  danceability
5207                Rebelution         0.975
2605                Gucci Mane         0.972
6916                      Tyga         0.968
3171               Jody Breeze         0.966
5395              Run–D.M.C.         0.965
430                   Ashnikko         0.964
431   Ashnikko, Princess Nokia         0.964
940                  Boy 8-Bit         0.962
3910                Lou Hayter         0.962
3214                    John K         0.958


In [87]:
# Get the largest 10 speechiness values
speechiness = df3.nlargest(10, 'speechiness')
print(speechiness[['Artist Name', 'speechiness']])


# Plot a bar chart, using plotly
px.bar(speechiness, 'Artist Name', 'speechiness', title='10 Most Speechiness Songs by Artist', color='speechiness', color_continuous_scale='ice')

# I honestly don't know what 'speechiness' mean but I assume it the opposite of a instrumental song.  I don't likethe speechiness scale on the right side, it's too squished.  I think it's because the 3rd Song's artist name is so long.  

                                            Artist Name  speechiness
2950                                       Ishmael Reed        0.960
234                                        Amiri Baraka        0.938
5148  Rajesh Khanna, Sumita Sanyal, Ramesh Deo, Seem...        0.829
238                                    Amitabh Bachchan        0.814
3643                                    Langston Hughes        0.806
4092                                    Margaret Walker        0.678
5283                                           Rob Frye        0.667
5817                                          Son House        0.658
4359                                       Moneybagg Yo        0.619
5227                  Rekha, Vinod Khanna, Ram P. Sethi        0.565


1. Look closely at the track team charts for energy and tempo. There’s a duplicate in each. How does that interrupt your bar chart? 

A. I went back and cleaned up the the duplicate I noticed in the energy chart.  I changed the tempo chart to display the Track instead of the Artist.  I have additional notes in comments directly in the script

2. After you complete the bar charts for all three audience types, look for overlap. For example, what artists are present in all three charts for the lounger type? Write those down as must-have songs for your song list. 

A. I actually think we did not pull the data correctly to easily see what songs/tracks overlap.  But we can see what artist overlap

Lounger:
* Carlos Cipa
* Anja Lechner
* Bach, Beetoven, and others were in two charts, but not all 3

Partier
* There is no overlap between the charts.  I think we'd have to pull in more than just the 10 largest/smallest to find overlap

NYT Bestseller
* There is no overlap between the charts.  I think we'd have to pull in more than just the 10 largest/smallest to find overlap

